`createTexture` fails with `dimension: '3d', format: 'r32float'`

I'm trying to update viv to the latest deck&luma and have an issue getting 3D textures to work.

https://github.com/hms-dbmi/viv/pull/805#issuecomment-2186381214

In xr-3d-layer.js, we call this (I believe that it is now not necessary to pass anything other than the format as a string, ie. 'r32float'):

    const attrs = getRenderingAttrs(); // attrs.format = 'r32float'
    const texture = this.context.device.createTexture({
      width,
      height,
      depth,
      dimension: '3d',
      data: attrs.cast?.(data) ?? data,
      format: attrs.format,
      mipmaps: false,
      sampler: {
        minFilter: 'linear',
        magFilterFilter: 'linear',
        addressModeU: 'clamp-to-edge',
        addressModeV: 'clamp-to-edge',
        addressModeW: 'clamp-to-edge'
      }
    });

luma.gl in the process of initialising the texture ends up calling `convertTextureFormatToGL(format)` in webgl-texture `setImageData3D()`, having already converted from string format to GL constant (33326). So it throws an "Unsupported texture format error". I briefly tried to patch my local version of luma to see how far it'd get if I overrode that, but didn't manage to apply that correctly.

To reproduce, you can clone my PR and use a sample with 3D data such as https://viv-demo.storage.googleapis.com/brain.pyramid.ome.tif, press the "show volumetric rendering" button, and select a resolution (probably the lowest, for testing).

@ibgreen any idea when it'll make it to a release version? Obviously you know better what else is in flux etc, but it'd definitely be great if it could.

That said - I tried adapting the sample to WebGPU

```js
  // luma.registerAdapters([webgl2Adapter]);
  luma.registerDevices([WebGPUDevice]);
  makeAnimationLoop(AnimationLoopTemplate, {type: "webgpu"}).start();
```

and am getting errors in `webgpu-texture.js` `setData()` with bad arguments to `new ImageData()` - both `this.width` and `this.height` are `1`, which they probably shouldn't be... so now I'm here and debugging, I'll look into that and make a PR if it makes sense to. Currently I still have issues with my IDE setup, showing errors like `Cannot find module '@luma.gl/engine' or its corresponding type declarations.`, but at least the code is running and editable.

--- a/modules/core/test/adapter/resources/texture.spec.ts
+++ b/modules/core/test/adapter/resources/texture.spec.ts
@@ -203,7 +203,7 @@ test('Texture#format creation with data', async t => {
   t.end();
 });
 
-test.only('Texture#dimension=3d,format=r32float', async t => {
+test('Texture#dimension=3d,format=r32float', async t => {
   for (const device of await getTestDevices()) {
     const texture = device.createTexture({
       width: 16,

--- a/modules/core/test/adapter/resources/texture.spec.ts
+++ b/modules/core/test/adapter/resources/texture.spec.ts
@@ -203,7 +203,29 @@ test('Texture#format creation with data', async t => {
   t.end();
 });
 
-test('Texture#copyExternaImage', async t => {
+test.only('Texture#dimension=3d,format=r32float', async t => {
+  for (const device of await getTestDevices()) {
+    const texture = device.createTexture({
+      width: 16,
+      height: 16,
+      depth: 16,
+      dimension: '3d',
+      format: 'r32float',
+      mipmaps: false,
+      sampler: {
+        minFilter: 'linear',
+        magFilterFilter: 'linear',
+        addressModeU: 'clamp-to-edge',
+        addressModeV: 'clamp-to-edge',
+        addressModeW: 'clamp-to-edge'
+      }
+    });
+    t.ok(texture, `${device.type}: Texture(dimension=3d,format=r32float) created`);
+    t.end();
+  }
+});
+
+test('Texture#copyExternalImage', async t => {
   for (const device of await getTestDevices()) {
     const texture = device.createTexture({
       width: 2,

